packages/data-access/src/services/RealTimeMutantsBlobService.ts

```diff
@@ -0,0 +1,73 @@
+import { ReportIdentifier } from '@stryker-mutator/dashboard-common';
+import { BlobServiceAsPromised } from './BlobServiceAsPromised.js';
+import { toBlobName } from '../utils.js';
+import { MutantResult } from 'mutation-testing-report-schema';
+
+// To make resource delete themselves automatically, this should be managed from within Azure:
+// https://learn.microsoft.com/en-us/azure/storage/blobs/lifecycle-management-overview?tabs=azure-portal
+export class RealTimeMutantsBlobService {
+  private static readonly CONTAINER_NAME = 'real-time-mutant-results';
+
+  #blobService: BlobServiceAsPromised;
+
+  constructor(blobService = new BlobServiceAsPromised()) {
+    this.#blobService = blobService;
+  }
+
+  public async createStorageIfNotExists() {
+    await this.#blobService.createContainerIfNotExists(
+      RealTimeMutantsBlobService.CONTAINER_NAME,
+      {}
+    );
+  }
+
+  public async createReport(id: ReportIdentifier) {
+    await this.#blobService.createAppendBlobFromText(
+      RealTimeMutantsBlobService.CONTAINER_NAME,
+      toBlobName(id),
+      ''
+    );
+  }
+
+  public async appendToReport(
+    id: ReportIdentifier,
+    mutants: Array<Partial<MutantResult>>
+  ) {
+    const blobName = toBlobName(id);
+    const data = mutants
+      .map((mutant) => `${JSON.stringify(mutant)}\n`)
+      .join('');
+
+    await this.#blobService.appendBlockFromText(
+      RealTimeMutantsBlobService.CONTAINER_NAME,
+      blobName,
+      data
+    );
+  }
+
+  public async getReport(
+    id: ReportIdentifier
+  ): Promise<Array<Partial<MutantResult>>> {
+    const data = await this.#blobService.blobToText(
+      RealTimeMutantsBlobService.CONTAINER_NAME,
+      toBlobName(id)
+    );
+
+    if (data === '') {
+      return [];
+    }
+
+    return data
+      .split('\n')
+      .slice(0, -1)
```

Comment:
Why is this done?

Reply:
Looks like this is used to remove the empty row. I should make that more clear or, even better, use filter instead :)